### PR TITLE
[receiver/simpleprometheus] fix lint for simpleprometheusreceiver

### PR DESCRIPTION
**Pull request by fatsheep9146**
Signed-off-by: Ziqi Zhao <zhaoziqi9146@gmail.com>

**Description:** <Describe what has changed.>
fix errcheck issue for simpleprometheusreceiver

**Link to tracking Issue:** <Issue number if applicable>
#9749

### CODE REVIEW

**Comment by jpkrohling** [comment]
The linter failed:

```
main module (github.com/open-telemetry/opentelemetry-collector-contrib/receiver/simpleprometheusreceiver) does not contain package github.com/open-telemetry/opentelemetry-collector-contrib/receiver/simpleprometheusreceiver/examples/federation/prom-counter
Check License finished successfully
running misspell -error
golangci-lint run --allow-parallel-runners
Error: main.go:65:19: Error return value of `logger.Sync` is not checked (errcheck)
	defer logger.Sync()
	                 ^
make[2]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/simpleprometheusreceiver/examples/federation/prom-counter'
```

**Comment by fatsheep9146** [comment]
> The linter failed:
> 
> ```
> main module (github.com/open-telemetry/opentelemetry-collector-contrib/receiver/simpleprometheusreceiver) does not contain package github.com/open-telemetry/opentelemetry-collector-contrib/receiver/simpleprometheusreceiver/examples/federation/prom-counter
> Check License finished successfully
> running misspell -error
> golangci-lint run --allow-parallel-runners
> Error: main.go:65:19: Error return value of `logger.Sync` is not checked (errcheck)
> 	defer logger.Sync()
> 	                 ^
> make[2]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/simpleprometheusreceiver/examples/federation/prom-counter'
> ```

Done

File: receiver/simpleprometheusreceiver/examples/federation/prom-counter/main.go [hunk 1]
```diff
@@ -63,7 +62,9 @@ func main() {
 	meter := initMeter()
 	// logging
 	logger, _ := zap.NewProduction()
-	defer logger.Sync()
+	defer func() {
```
Comment (jpkrohling) [inline receiver/simpleprometheusreceiver/examples/federation/prom-counter/main.go:65]: Why do you need to wrap this in a func?

Comment (fatsheep9146) [reply receiver/simpleprometheusreceiver/examples/federation/prom-counter/main.go:65]: No special reason, just follow other issues' way

https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/12592/files#diff-d4418701fb60e9de7b034cb0351c95e197998f612cd322e2d4272c342055f1f7R111-R113

Comment (dmitryax) [reply receiver/simpleprometheusreceiver/examples/federation/prom-counter/main.go:65]: I believe there is no other way to satisfy errcheck linter

**Review by dmitryax — APPROVED** [review]

**Pull request merged**